hearth/lib/hearth/telemetry/span_status.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+module Hearth
+  module Telemetry
+    # Represents the status of a finished span.
+    class SpanStatus
+      class << self
+        private :new
+
+        # Returns a newly created {SpanStatus} with code == UNSET
+        #  and an optional description.
+        #
+        # @param [String] description
+        # @return [SpanStatus]
+        def unset(description = '')
+          new(UNSET, description: description)
+        end
+
+        # Returns a newly created {SpanStatus} with code == OK
+        #  and an optional description.
+        #
+        # @param [String] description
+        # @return [SpanStatus]
+        def ok(description = '')
+          new(OK, description: description)
+        end
+
+        # Returns a newly created {SpanStatus} with code == ERROR
+        #  and an optional description.
+        #
+        # @param [String] description
+        # @return [SpanStatus]
+        def error(description = '')
+          new(ERROR, description: description)
+        end
+      end
+
+      def initialize(code, description: '')
+        @code = code
+        @description = description
+      end
+
+      attr_reader :code, :description
+
+      # The operation completed successfully.
+      OK = 0
```

Comment:
Are these constants ever checked anywhere? Should these be private (by not existing)?

Reply:
Yes, these constants are intended for use. I use them [here](https://github.com/smithy-lang/smithy-ruby/blob/dc2072409e1d4e76e28aa44ba4eb3960bd063a4f/codegen/projections/rails_json/lib/rails_json/client.rb#L102) for each operation. 

Customers who intend to implement their own telemetry provider could utilize these as well.

Reply:
That's different. I'm talking about for OK, UNSET, and ERROR. In your linked code I see `Hearth::Telemetry::SpanKind::CLIENT`

Reply:
Oops. I was looking at the wrong tracer component. 

Hmm. I was thinking of the following usage:
```
class CustomSpan
  def status=(status)
    return if status.code == Hearth::Telemetry::SpanStatus::UNSET
    ...
  end
end

```
I'm guessing OpenTelemetry-Ruby did this for readability instead of figuring what code does what.